### PR TITLE
docs: update vimdoc parser

### LINKED ISSUE #49: parse error: options.txt
Found `tree-sitter-vimdoc` parse error at: https://neovim.io/doc/user/options.html

Context:

```
	   "number"	display signs in the 'number' column. If the number
			column is not present, then behaves like "auto".

	Note regarding 'orphaned signs': with signcolumn numbers higher than
	1, deleting lines will also remove the associated signs automatically,
	in contrast to the default Vim behavior of keeping and grouping them.
	This is done in order for the signcolumn appearance not appear weird
```

Umm. What just happened? Where am I?

### LINKED ISSUE #50: two lines are combined as one: options.txt
Context:

```
If you want to include a ':' in a set command precede it with a '\'.  The
backslash in front of the ':' will be removed.  Example:
```

The code block below are combined with the line following it (`This sets the 'fillchars' option`). 

Also in context for option `filetype`

```
When a dot appears in the value then this separates two filetype
	names.  Example:
```

Same issue.

### LINKED ISSUE #51: parse error: options.txt
Found `tree-sitter-vimdoc` parse error at: https://neovim.io/doc/user/options.html

Context:

```
	to check what is following.  E.g. for Javascript, if a function is
	defined with "func_name = function(args)": >
		^\s*\ze\i\+\s*=\s*function(
<	If the function is defined with "func_name : function() {...": >
	        ^\s*\ze\i\+\s*[:]\s*(*function\s*(
<	When using the ":set" command, you need to double the backslashes!
	To avoid that use `:let` with a single quote string: >
```

The 3rd code block is not rendered correctly as code block.

### PR DESCRIPTION
**Pull request by justinmk**
Remove the user-manual ToC from help.txt, because:
1. it duplicates usr_toc.txt
2. it is not what most readers are looking for in the main help page.

fix https://github.com/neovim/tree-sitter-vimdoc/issues/49
fix https://github.com/neovim/tree-sitter-vimdoc/issues/50
fix https://github.com/neovim/tree-sitter-vimdoc/issues/51

### CODE REVIEW

File: runtime/doc/help.txt [hunk 1]
```diff
@@ -38,169 +38,126 @@ Get specific help:  It is possible to go directly to whatever you want help
...
-|help.txt|	overview and quick reference (this file)
-|helphelp.txt|	about using the help files
-|index.txt|	alphabetical index of all commands
-|tips.txt|	various tips on using Vim
-|message.txt|	(error) messages and explanations
-|develop.txt|	development of Nvim
-|debug.txt|	debugging Vim itself
-|uganda.txt|	Vim distribution conditions and what to do with your money
+					*reference_toc* *doc-file-list* *Q_ct*
+Meta ~
```
Comment (clason) [inline runtime/doc/help.txt:42]: What's the difference between "Meta" and "General"?

Might be confusing; maybe simpler to have a combined list but broken up into digestible chunks by empty lines? (Then people will at least not be distracted by trying to interpret the meaning of the section headings.)

File: runtime/doc/help.txt [hunk 2]
```diff
@@ -38,169 +38,126 @@ Get specific help:  It is possible to go directly to whatever you want help
...
+|nvim|			Transitioning from Vim
```
Comment (clason) [inline runtime/doc/help.txt:43]:
```suggestion
|news.txt|      News and notable changes in the latest release
|nvim|			Transitioning from Vim
```

File: runtime/doc/help.txt [hunk 3]
```diff
@@ -38,169 +38,126 @@ Get specific help:  It is possible to go directly to whatever you want help
...
+|tutor|			30-minute interactive course for beginners
+|copying|		About copyrights
+|iccf|			Helping poor children in Uganda
+|sponsor|		Sponsor Vim development, become a registered Vim user
+|www|			Vim on the World Wide Web
+|bugs|			Where to send bug reports
+|support|		Supported platforms
+
+General ~
+|intro.txt|		Introduction to Vim; notation used in help files
```
Comment (clason) [inline runtime/doc/help.txt:56]: Nitpick: Would it be possible to homogenize (drop `.txt` everywhere)?

Comment (justinmk) [reply runtime/doc/help.txt:59]: I did that where possible. Some of these don't have non-txt tags. Not sure
"intro" deserves its own tag.

Can revisit this later. Much of these will be reworked anyway.

File: runtime/doc/help.txt (hunk 2 above)
Comment (clason) [reply runtime/doc/help.txt:48]: Yeah, I commented on the state where you removed half the universe (including this entry). Now that they're back, I'm not concerned about the location.

**Pull request merged**